src/multi_vector_simulator/E1_process_results.py

```diff
@@ -554,24 +554,42 @@ def add_info_flows(settings, dict_asset, flow):
     )
 
 
-def convert_demand_to_dataframe(dict_values):
+def convert_demand_to_dataframe(dict_values, sector_demands=None):
     """Dataframe used for the demands table of the report
 
     Parameters
     ----------
     dict_values: dict
         output values of MVS
 
+    sector_demands: str
+        Name of the sector of the energy system whose demands must be returned as a df by this function
+        Default: None
+
     Returns
     -------
     :class:`pandas.DataFrame<frame>`
 
     """
-    # Creating a dataframe for the demands
-    demands = dict_values[ENERGY_CONSUMPTION]
 
-    # Removing all columns that are not actually from demands
+    # Make a dict which is a sub-dict of the JSON results file with only the consumption components of the energy system
+    demands = dict(dict_values[ENERGY_CONSUMPTION])
```

Comment:
you don't want to use `copy.deepcopy`?

Reply:
The folks at stackoverflow suggested this on one of the threads, so I used this. Can modify, but it seems to work fine

Reply:
Stackoverflow is great for patches but can also lead to bugs which are super hard to find in the future. I know deep copy allocates memory for a new copy of the dict, the other method we would need to test it. I changed it (and checked the report was similar)